filebeat/include/fields.go

```diff
@@ -0,0 +1,20 @@
+// ${ES_BEATS}/dev-tools/cmd/asset/asset.go
+// MACHINE GENERATED BY THE ABOVE COMMAND; DO NOT EDIT
+
+package include
+
+import (
+	"encoding/base64"
+
+	"github.com/elastic/beats/libbeat/assets"
+)
+
+func init() {
+	if err := assets.AddFields("fields.yml", Asset); err != nil {
```

Comment:
If two Beats were combined into the same process would this still work?

Reply:
The `AddFields` would work as currently the name is not used at all internally. What would cause problems is that the fields from libbeat would be duplicated. For now I kept the fields.yml exactly as it is and a 1-1 copy is added to each file. For some long terms ideas like splitting up for libbeat or modules, see PR description.

Reply:
I was thinking that you would want the fields from each Beat to be stored independently such that each template (from `asset.GetFields`) contains only the fields for that Beat. But maybe that is moot if all beats in one process share a single output (not sure how that would work).

Reply:
It's definitively a valid point but I would ignore it for now.